pkg/controller/cyclenodestatus/controller.go

```diff
@@ -114,6 +115,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	rm := cyclecontroller.NewResourceManager(
 		r.mgr.GetClient(),
 		r.rawClient,
+		*http.DefaultClient,
```

Comment:
I was going to say don't use DefaultClient because it has no timeout set, but looks like it's not used for the cns controller here. I see you've set the timeout on her cnr client 👍

However as the client is optional, not used for cns, it might be better to have it as a pointer as pass nil when not expected. Alternatively just make a client with a timeout here too so there's no gotchas in the future.